rfcs/0032-ckb-vm-version-selection/0032-ckb-vm-version-selection.md

```diff
@@ -1 +1,81 @@
+
+After ckb2021 is activated, CKB nodes must choose the CKB VM version for each script group. The allowed values for the `hash_type` field in the lock and type script are 0, 1, and 2. Cells are sorted into different groups if they have different `hash_type`. According to the value of `hash_type`:
+
+* When the `hash_type` is 0, the script group matches code via data hash and will run the code using the CKB VM version 0.
+* When the `hash_type` is 1, the script group matches code via type script hash and will run the code using the CKB VM version 1.
+* When the `hash_type` is 2, the script group matches code via data hash and will run the code using the CKB VM version 1.
+
+| `hash_type` | matches by       | VM version |
+| ----------- | ---------------- | ---------- |
+| 0           | data hash        | 0          |
+| 1           | type script hash | 1          |
+| 2           | data hash        | 1          |
+
+See following sections about VM versions.
+
+The transaction is considered invalid if any `hash_type` is not in the allowed values 0, 1, and 2.
+
+Because the VM selection algorithm depends on which epoch the transaction belongs to, it is not deterministic for transactions still in the memory pool. The CKB node must run two versions of transaction relay protocols, one for the current CKB version C and another for the next fork version N.
+
+* Before the fork is activated, CKB node must relay transactions via relay protocol C and must drop all messages received via protocol N.
+* After the fork is activated, CKB node must relay transactions via relay protocol N and must drop all messages received via protocol C.
+
+The relay protocol C will be dropped after the fork succeeds. See [rfc35] for details.
+
+[rfc35]: ../0035-ckb2021-p2p-protocol-upgrade/0035-ckb2021-p2p-protocol-upgrade.md
+
+When a new block is appended to the chain and the fork is activated, or a block is rolled back and the fork is deactivated, the CKB node must rerun the verification on all the transactions in the pool.
+
+In the [Rust implementation of CKB](https://github.com/nervosnetwork/ckb) provided by Nervos Foundation, the `hash_type` is returned as an enum. Now it has three allowed values:
+
+* 0: "data"
+* 1: "type"
+* 2: "data1"
+
+### CKB VM Versions
+
+The VM version 0 deployed with the Lina and Aggron genesis block is version 0.
+
+The VM version 1 will be deployed along ckb2021. The [rfc33] highlights the changes in this version, and [rfc34] lists the new syscalls only available since version 1.
+
+[rfc33]: ../0033-ckb-vm-version-1/0033-ckb-vm-version-1.md
+[rfc34]: ../0034-vm-syscalls-2/0034-vm-syscalls-2.md
```

Comment:
RFC-0035 should be listed here, too.
Those changes are all due to this RFC.

RFC-0032, RFC-0033, RFC-0034 and RFC-0035 have to be enabled together.

Reply:
Resolved by dd5f26a02d3db1659f86c42bdbf52c49975a68bb